### PR TITLE
Test net8

### PR DESCRIPTION
**Pull request by sm6srw**
Test net8 build

### Purpose

(FILL ME IN) This section describes why this PR is here. Usually it would include a reference to the tracking task that it is part or all of the solution for.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/build.xml [hunk 1]
```diff
@@ -5,14 +5,14 @@
 <PropertyGroup>
     <Solution>Dynamo.All.sln</Solution>
     <Platform>Any CPU</Platform>
-    <DotNet>net6.0</DotNet>
+    <DotNet>net8.0</DotNet>
```
Comment (pinzart90) [inline src/build.xml:8]: you can revert this and just run `msbuild build.xml /p:DotNet=net8.0` in the jenkins file to test it

File: test/DynamoCoreWpfTests/PackageManager/PackageManagerViewExtensionTests.cs [hunk 2]
```diff
@@ -148,7 +148,7 @@ public void PackageManagerViewExtensionHasCorrectNumberOfRequestedExtensions()
         }
 
         [Test]
-        async public void LateLoadedViewExtensionsHaveMethodsCalled()
```
Comment (pinzart90) [inline test/DynamoCoreWpfTests/PackageManager/PackageManagerViewExtensionTests.cs:151]: removed async.
Not sure why it was needed

File: src/build.xml (hunk 1 above)
Comment (sm6srw) [reply src/build.xml:8]: Yes, I will revert this before merge but leave this open for a while as a test branch for the BRE team to use.

**Comment by sm6srw** [comment]
Looks like we got some new files now.

**Comment by pinzart90** [comment]
> Looks like we got some new files now.

Yeah, without the `test` defaults, we now have to be explicit.
So probably all Moq package references need to be marked ExcludeAssets="runtime"

File: src/Config/CS_SDK.props [hunk 3]
```diff
@@ -67,6 +65,7 @@
   <PropertyGroup>
     <BaseIntermediateOutputPath>int\$(TargetFramework)\</BaseIntermediateOutputPath>
     <IntermediateOutputPath>obj\$(TargetFramework)\</IntermediateOutputPath>
+    <TestDependenciesPath>$(OutputPath)\..\..\..\test\test_dependencies</TestDependenciesPath>
```
Comment (aparajit-pratap) [inline src/Config/CS_SDK.props:68]: I see some of the packages required for testing are copied to this path but how is it being used?

File: test/DynamoCoreTests/DynamoCoreTests.csproj [hunk 4]
```diff
@@ -37,16 +37,16 @@
         </PackageReference>
 
         <!--moq and deps - copied to test_dependencies folder-->
-        <PackageReference Include="Moq" Version="4.18.4" GeneratePathProperty="true" />
-        <PackageReference Include="Castle.Core" Version=" 5.1.1" GeneratePathProperty="true" />
+        <PackageReference Include="Moq" Version="4.18.4" ExcludeAssets="runtime" GeneratePathProperty="true" />
+        <PackageReference Include="Castle.Core" Version=" 5.1.1" ExcludeAssets="runtime" GeneratePathProperty="true" />
```
Comment (aparajit-pratap) [inline test/DynamoCoreTests/DynamoCoreTests.csproj:41]: What happens if you don't exclude these assets? Are they all copied to the test_dependencies path in that case?

File: test/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<Project>
+  <PropertyGroup>
+    <TestProject>true</TestProject>
```
Comment (pinzart90) [inline test/Directory.Build.props:3]: This will mark all projects in the test subfolder as being TestProjects (TestProject == true)

Comment (pinzart90) [reply test/Directory.Build.props:3]: seems like this keeps the changes to a minimum

**Comment by pinzart90** [comment]
@sm6srw I think we should adapt this PR for master and get it in.

**Comment by sm6srw** [comment]
> @sm6srw I think we should adapt this PR for master and get it in.

Yes

**Review by pinzart90 — APPROVED** [review]

**Pull request merged**